src/pymmcore_widgets/mda/_core_mda.py

```diff
@@ -0,0 +1,251 @@
+    QBoxLayout,
+    QCheckBox,
+    QFileDialog,
+    QGridLayout,
+    QGroupBox,
+    QHBoxLayout,
+    QLabel,
+    QLineEdit,
+    QPushButton,
+    QWidget,
+)
+from superqt.fonticon import icon
+
+from pymmcore_widgets.useq_widgets import MDASequenceWidget
+
+from ._core_positions import CoreConnectedPositionTable
+from ._core_z import CoreConnectedZPlanWidgert
+
+if TYPE_CHECKING:
+    from typing import TypedDict
+
+    from useq import MDASequence
+
+    class SaveInfo(TypedDict):
+        save_dir: str
+        file_name: str
+        split_positions: bool
+        should_save: bool
+
+
+class MDAWidget(MDASequenceWidget):
+    """Widget for running MDA experiments, connecting to a MMCorePlus instance."""
+
+    def __init__(
+        self, *, parent: QWidget | None = None, mmcore: CMMCorePlus | None = None
+    ) -> None:
+        # create a couple core-connected variants of the tab widgets
+        self._mmc = mmcore or CMMCorePlus.instance()
+        position_wdg = CoreConnectedPositionTable(1, self._mmc)
+        z_wdg = CoreConnectedZPlanWidgert(self._mmc)
+
+        super().__init__(parent=parent, position_wdg=position_wdg, z_wdg=z_wdg)
+
+        self.save_info = _SaveGroupBox(parent=self)
+        self.control_btns = _MDAControlButtons(self._mmc, self)
+
+        # -------- initialize -----------
+
+        self.z_plan.setSuggestedStep(_guess_NA(self._mmc) or 0.5)
+        # TODO: connect objective change event to update suggested step
+        self._update_channel_groups()
+
+        # ------------ layout ------------
+
+        layout = cast("QBoxLayout", self.layout())
+        layout.insertWidget(0, self.save_info)
+        layout.addWidget(self.control_btns)
+
+        # ------------ connect signals ------------
+
+        self.control_btns.run_btn.clicked.connect(self._on_run_clicked)
+        self.control_btns.pause_btn.released.connect(self._mmc.mda.toggle_pause)
+        self.control_btns.cancel_btn.released.connect(self._mmc.mda.cancel)
+        self._mmc.mda.events.sequenceStarted.connect(self._on_mda_started)
+        self._mmc.mda.events.sequenceFinished.connect(self._on_mda_finished)
+        self._mmc.events.channelGroupChanged.connect(self._update_channel_groups)
+
+        self.destroyed.connect(self._disconnect)
+
+    def value(self) -> MDASequence:
+        """Set the current state of the widget."""
+        val = super().value()
+        meta: dict = val.metadata.setdefault("pymmcore_widgets", {})
+        meta.update(self.save_info.value())
+        return val
+
+    def setValue(self, value: MDASequence) -> None:
+        """Get the current state of the widget."""
+        super().setValue(value)
+        self.save_info.setValue(value.metadata.get("pymmcore_widgets", {}))
+
+    # ------------------- private API ----------------------
+
+    def _update_channel_groups(self) -> None:
+        ch = self._mmc.getChannelGroup()
+        self.channels.setChannelGroups({ch: self._mmc.getAvailableConfigs(ch)})
```

Comment:
I think it is safer if we do something like this here so that we see the combobox all the time. What do you think?

```suggestion
        self.channels.setChannelGroups(
            {
                group_name: self._mmc.getAvailableConfigs(group_name)
                for group_name in self._mmc.getAvailableConfigGroups()
            }
        )
        if ch := self._mmc.getChannelGroup():
            self.channels._group_combo.setCurrentText(ch)
```

Reply:
i was kinda hoping to avoid the added complexity.  but if `getChannelGroup()` returns nothing, then I agree it would be good to show everything.